src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -165,285 +166,345 @@ subroutine ocn_tracer_hmix_redi_tend(meshPool, scratchPool, layerThicknessEdge,
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term1',config_disable_redi_horizontal_term1)
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term2',config_disable_redi_horizontal_term2)
-      call mpas_pool_get_config(ocnConfigs, 'config_disable_redi_horizontal_term3',config_disable_redi_horizontal_term3)
-
-      !
-      ! COMPUTE the extra terms arising due to mismatch between the constant coordinate surfaces and the
-      ! isopycnal surfaces.
-      !
-
-      call mpas_pool_get_field(scratchPool, 'gradTracerEdge', gradTracerEdgeField)
-      call mpas_pool_get_field(scratchPool, 'gradTracerTopOfEdge', gradTracerTopOfEdgeField)
-      call mpas_pool_get_field(scratchPool, 'gradHTracerSlopedTopOfCell', gradHTracerSlopedTopOfCellField)
-      call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfCell', dTracerdZTopOfCellField)
-      call mpas_pool_get_field(scratchPool, 'dTracerdZTopOfEdge', dTracerdZTopOfEdgeField)
-      call mpas_pool_get_field(scratchPool, 'areaCellSum', areaCellSumField)
-
-      call mpas_allocate_scratch_field(gradTracerEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(gradTracerTopOfEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(gradHTracerSlopedTopOfCellField, .true., .false.)
-      call mpas_allocate_scratch_field(dTracerdZTopOfCellField, .true., .false.)
-      call mpas_allocate_scratch_field(dTracerdZTopOfEdgeField, .true., .false.)
-      call mpas_allocate_scratch_field(areaCellSumField, .true., .false.)
-
-      gradTracerEdge => gradTracerEdgeField % array
-      gradTracerTopOfEdge => gradTracerTopOfEdgeField % array
-      gradHTracerSlopedTopOfCell => gradHTracerSlopedTopOfCellField % array
-      dTracerdZTopOfCell => dTracerdZTopOfCellField % array
-      dTracerdZTopOfEdge => dTracerdZTopOfEdgeField % array
-      areaCellSum => areaCellSumField % array
-
       nCells = nCellsArray( size(nCellsArray) )
       nEdges = nEdgesArray( size(nEdgesArray) )
 
-      !$omp do schedule(runtime)
+      allocate(minimumVal(nTracers))
+      allocate(redi_term2_edge(nTracers,nVertLevels,nEdges))
+      allocate(redi_term3_topOfCell(nTracers,nVertLevels+1,nCells))
+
+      nCells = nCellsArray(1)
+      nCellsP1 = nCellsArray( size(nCellsArray) ) + 1
+      redi_term2(:,:,:) = 0.0_RKIND
+      redi_term2_edge(:,:,:) = 0.0_RKIND
+      redi_term3_topOfCell(:,:,:) = 0.0_RKIND
+
+      allocate(fluxRediZTop(nTracers,nVertLevels+1))
+      redi_term3(:,:,:) = 0.0_RKIND
+      redi_term1(:,:,:) = 0.0_RKIND
+      ! Term 1: this is the "standard" horizontal del2 term, but with RediKappa coefficient.
+      ! \kappa_2 \nabla \phi on edge
+      ! mrp performance: merge term 1 into main cell loop later
+      !$omp do schedule(runtime) private(invAreaCell, i, iEdge, cell1, cell2, iCellSelf) &
+      !$omp& private(r_tmp, coef, kappaRediEdgeVal, tracer_turb_flux, flux, k, iTr, flux_term2) 
```

Comment:
Same thing here and in this file, if you could check my choice of private variables.

Reply:
An update for this file:
```
azamatm@thetalogin2:~/repos/E3SM-clone/components/mpas-source> git diff src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F
diff --git a/src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F b/src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F
index 6d8141ff..06ec35bc 100644
--- a/src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F
+++ b/src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F
@@ -188,8 +188,8 @@ contains
       ! Term 1: this is the "standard" horizontal del2 term, but with RediKappa coefficient.
       ! \kappa_2 \nabla \phi on edge
       ! mrp performance: merge term 1 into main cell loop later
-      !$omp do schedule(runtime) private(invAreaCell, i, iEdge, cell1, cell2, iCellSelf) &
-      !$omp& private(r_tmp, coef, kappaRediEdgeVal, tracer_turb_flux, flux, k, iTr, flux_term2) 
+      !$omp do schedule(runtime) private(invAreaCell, i, iEdge, cell1, cell2, iCellSelf, &
+      !$omp& r_tmp, coef, kappaRediEdgeVal, tracer_turb_flux, flux, k, iTr, flux_term2, dTracerDx, flux_term3)
       do iCell = 1, nCells
          invAreaCell = 1.0_RKIND / areaCell(iCell)
          fluxRediZTop(:,:) = 0.0_RKIND
@@ -369,7 +369,7 @@ contains
      !$omp end do
 
      !now go back and reapply all tendencies
-     !$omp do schedule(runtime) private(invAreaCell, k, iTr, i)
+     !$omp do schedule(runtime) private(invAreaCell, k, iTr, i, iEdge)
      do iCell=1,nCells
        invAreaCell = 1.0_RKIND / areaCell(iCell)
         do k = 1, maxLevelCell(iCell)
```